src/engine/controls/clockcontrol.cpp

```diff
@@ -204,3 +214,44 @@ void ClockControl::updateIndicators(const double dRate,
     }
     m_lastEvaluatedPosition = currentPosition;
 }
+
+void ClockControl::updateBeatCounter(mixxx::BeatsPointer pBeats, 
+    mixxx::audio::FramePos currentFramePos) {
+
+    QList<mixxx::audio::FramePos> cuesFromCurrentPosition =
+            QList<mixxx::audio::FramePos>();
+
+    //Iterate through current Track cues and create a list with the FramePos of the ones
+    // that are after the current play position. We add them ordered in the new list
+    for (int i = 0; i < m_pTrackCues.count(); ++i) {
+        CuePointer cue = m_pTrackCues[i];
+        mixxx::audio::FramePos cueFramePos = cue->getPosition();
+        if (cueFramePos.isValid()) {
+            if (cueFramePos >= currentFramePos) {
+                if (cuesFromCurrentPosition.isEmpty()) {
+                    cuesFromCurrentPosition.append(cueFramePos);
```

Comment:
Since ClockControl::updateBeatCounter is called in a tight loop, we should find a solution that does nor required memory allocation. Can we get rid of cuesFromCurrentPosition, we use only the first element below anyway?

Reply:
Thanks, I am still getting my head around what is tight and what is decoupled in mixxx
My initial thought for the feature was to be able to have the counter for *all* the next cues to help with multi-drop mixes, but I believe in practice you may only use one or two of the upcoming cues. Will definitely move to a single cue implementation and think if / how the multi-drop feature could be better implemented